com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -186,6 +202,37 @@ void OnEnable()
             LazyInitialize();
         }
 
+
+        public void OnBeforeSerialize()
+        {
+            if (maxStep != agentParameters.maxStep && !hasUpgradedFromAgentParameters)
+            {
+                maxStep = agentParameters.maxStep;
+                m_MarkDirtFromUpgrade = true;
+            }
+        }
+
+        public void OnAfterDeserialize()
+        {
+            if (maxStep != agentParameters.maxStep && !hasUpgradedFromAgentParameters)
+            {
+                Debug.Log("Upgrading from agentParameters.maxStep to this.maxStep.  Please save your project to keep these changes");
```

Comment:
After loading a scene where I got these logs, I tried saving (Cmd-S and File > Save Project) and neither one modified any files. Tried playing the scene too (wasn't sure when Awake triggered) but no files looked modified according to `git status`.

Reply:
hmm, I'll doublecheck this.

Reply:
removed it